packages/core/src/textures/resources/createResource.js

```diff
@@ -3,22 +3,28 @@ import SVGResource from './SVGResource';
 import CanvasResource from './CanvasResource';
 import VideoResource from './VideoResource';
 
+const videosTypes = ['mp4', 'm4v', 'webm', 'ogg', 'ogv', 'h264', 'avi', 'mov'];
+
 export default function createResource(source)
 {
     if (typeof source === 'string')
     {
-        // check if its a video..
-        if (source.match(/\.(mp4|webm|ogg|h264|avi|mov)$/))
-        {
-            return new VideoResource.fromUrl(source);
-            // video!
-            // return Texture.fromVideoUrl(source);
-            // return SVGResource.from(url);
-        }
-        else if (source.match(/\.(svg)$/))
+        // search for file extension: period, 3-4 chars, then ?, # or EOL
+        const result = (/\.(\w{3,4})(?:$|\?|#)/i).exec(source);
```

Comment:
Using the same RegExp from v4's BaseTexture for resolving the file extension. This solves issues with files ending in "?...", "#...", as well as non-lowercasing.

Reply:
nice. merge it.